imap_data_access/file_validation.py

```diff
@@ -444,11 +539,142 @@ def construct_path(self) -> Path:
         # Use the file suffix to determine the directory structure
         # IMAP_DATA_DIR/spice/<subdir>/filename
         return spice_dir / subdir / self.filename
 
+    @staticmethod
+    def _handle_doy(part, components):
+        """Parse a YYYY_DOY string and assigns it to start_date or end_date."""
+        value = components[part]
+        try:
+            doy_as_datetime = datetime.strptime(value, "%Y_%j")
+        except ValueError:
+            return "Invalid day of year format. Please use YYYY_DOY format."
+
+        if "start" in part:
+            components["start_date"] = doy_as_datetime
+        elif "end" in part:
+            components["end_date"] = doy_as_datetime
+        del components[part]
+
+    @staticmethod
+    def _handle_date(part, components):
+        """Parse a YYYYMMDD string and stores a datetime object back in components."""
+        value = components[part]
+        try:
+            components[part] = datetime.strptime(value, "%Y%m%d")
+        except ValueError:
+            return "Invalid date format. Please use YYYYMMDD format."
+
+    @staticmethod
+    def _handle_type(part, components):
+        """Map the 'type' part to its corresponding SPICE type."""
+        value = components[part]
+        try:
+            components[part] = _SPICE_TYPE_MAPPING[value]
+        except KeyError:
+            return f"SPICE type '{value}' not found in the approved SPICE types."
+
+    @staticmethod
+    def _handle_start_year(part, components):
+        """Set the year and also computes a start_date of January 1 for that year."""
+        value = components[part]
+        year_int = int(value)
+        components[part] = int(year_int)
+        try:
+            components["start_date"] = datetime(year_int, 1, 1)
+        except ValueError:
+            return "Invalid year. Please ensure the year is not 0."
+        del components[part]
+
+    @staticmethod
+    def _handle_version(part, components):
+        """Parse version into an integer."""
+        value = components[part]
+        components[part] = int(value)
+
+    @staticmethod
+    def _spice_parts_handler(components):
+        """Validate and transform SPICE file compents.
+
+        Parameters
+        ----------
+        components : dict
+            Dictionary containing components of the file.
+
+        Returns
+        -------
+        components : dict
+            Dictionary containing components, validated and transformed.
+        """
+        parts_list = list(components.keys())
+        transforms = {
+            "start_year_doy": SPICEFilePath._handle_doy,
+            "end_year_doy": SPICEFilePath._handle_doy,
+            "start_date": SPICEFilePath._handle_date,
+            "end_date": SPICEFilePath._handle_date,
+            "type": SPICEFilePath._handle_type,
+            "start_year": SPICEFilePath._handle_start_year,
+            "version": SPICEFilePath._handle_version,
+        }
```

Comment:
My suggestion is to not be too fancy and just manually handle these here. I personally don't think we need to report multiple error strings and the like. That can be follow up if people really like super-validation, but sometimes just failing multiple times is fine IMO, especially if it simplifies the overall code structure.

```suggestion
        if "start_date" in components:
            components["start_date"] = datetime.strptime(components["start_date"], "%Y%m%d")
        if "start_year_doy" in components:
            components["start_date"] = datetime.strptime(components.pop("start_year_doy"), "%Y_%j")
```